Unix/server/server.c

```diff
@@ -265,6 +384,13 @@ int servermain(int argc, const char* argv[])
     /* Extract command-line options a second time (to override) */
     GetCommandLineOptions(&argc, argv);
 
+    engine_envp = _DuplicateEnvp(envp);
+    if (!engine_envp)
+    {
+        info_exit(ZT("probable out of memory\n"));
+        exit(2);
```

Comment:
server main is sometimes called from unittests. You probably need to return a return code rather than exiting the process, and unittest process